### PR TITLE
Add description of Apify-haystack integration

### PR DESCRIPTION
**Pull request by jirispilka**
Add Apify-Haystack integration

This integration allows orchestrating web crawling within haystack framework, e.g. for RAG application.

Apify is a platform for web scraping, data extraction, and web automation tasks.
It provides serverless applications called Actors for different tasks, like crawling websites, scraping Facebook etc.

### CODE REVIEW

File: integrations/apify.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+In the examples below, specify `apify_api_token` and run the script.
+
+
+### ApifyDatasetFromActorCall on its own
+
+
+Use Apify's [Website Content Crawler](https://apify.com/apify/website-content-crawler) Actor to crawl a website, scrape text content, and convert it into Haystack Documents.
+
+In the example below, the text content is extracted from https://haystack.deepset.ai/. 
+You can control the number of crawled pages using `maxCrawlPages` parameter. For the detailed overview of the parameters, please refer to the [Website Content Crawler](https://apify.com/apify/website-content-crawler/input-schema).
```
Comment (BraniganLee) [inline integrations/apify.md:68]:
```suggestion
You can control the number of crawled pages using `maxCrawlPages` parameter. For a detailed overview of the parameters, please refer to [Website Content Crawler](https://apify.com/apify/website-content-crawler/input-schema).
```

File: integrations/apify.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+actor_id = "apify/website-content-crawler"
+run_input = {
+    "maxCrawlPages": 3,  # limit the number of pages to crawl
+    "startUrls": [{"url": "https://haystack.deepset.ai/"}],
+}
+
+
+def dataset_mapping_function(dataset_item: dict) -> Document:
+    """Convert Apify dataset item to Haystack Document
```
Comment (BraniganLee) [inline integrations/apify.md:93]:
```suggestion
    """Convert an Apify dataset item to a Haystack Document
```

File: integrations/apify.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    
+    The Website Content Crawler Actor returns a dataset with the following output fields:
```
Comment (BraniganLee) [inline integrations/apify.md:95]:
```suggestion
   Website Content Crawler returns a dataset with the following output fields:
```

File: integrations/apify.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+    return Document(content=dataset_item.get("text"), meta={"url": dataset_item.get("url")})
+
+
+actor = ApifyDatasetFromActorCall(
+    actor_id=actor_id,
+    run_input=run_input,
+    dataset_mapping_function=dataset_mapping_function,
+    apify_api_token=apify_api_token,
+)
+print(f"Calling the Apify actor {actor_id} ... crawling will take some time ...")
```
Comment (BraniganLee) [inline integrations/apify.md:110]:
```suggestion
print(f"Calling the Apify Actor {actor_id} ... crawling will take some time ...")
```

File: integrations/apify.md [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+
+actor_id = "apify/website-content-crawler"
+run_input = {
+    "maxCrawlPages": 1,  # limit the number of pages to crawl
+    "startUrls": [{"url": "https://haystack.deepset.ai/"}],
+}
+
+
+def dataset_mapping_function(dataset_item: dict) -> Document:
+    """Convert Apify dataset item to Haystack Document
```
Comment (BraniganLee) [inline integrations/apify.md:161]:
```suggestion
    """Convert an Apify dataset item to a Haystack Document
```

File: integrations/apify.md [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+    
+    The Website Content Crawler Actor returns a dataset with the following output fields:
```
Comment (BraniganLee) [inline integrations/apify.md:163]:
```suggestion
   Website Content Crawler returns a dataset with the following output fields:
```

**Review by BraniganLee — CHANGES_REQUESTED** [review]

File: integrations/apify.md [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+You need to have an Apify account and an Apify API token to run this example.
+You can start with a free account at [Apify](https://apify.com/) and get your [Apify API token](https://docs.apify.com/platform/integrations/api#api-token).
+
+In the examples below, specify `apify_api_token` and run the script.
+
+
+### ApifyDatasetFromActorCall on its own
+
+
+Use Apify's [Website Content Crawler](https://apify.com/apify/website-content-crawler) to crawl a website, scrape text content, and convert it to Haystack Documents.
```
Comment (bilgeyucel) [inline integrations/apify.md:65]: Feel free to mention Apify Store here with a sentence like this: 
```suggestion
Use Apify's [Website Content Crawler](https://apify.com/apify/website-content-crawler) to crawl a website, scrape text content, and convert it to Haystack Documents. You can browse other Actors in [Apify Store](https://apify.com/store)
```

File: integrations/apify.md [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+print("You can visit https://console.apify.com/actors/runs to monitor the progress")
+
+response = pipe.run({"embedder": {"text": question}, "prompt_builder": {"question": question}})
+
+print(f"question: {question}")
+print(f"answer: {response['llm']['replies'][0]}")
+
+# Other questions
+examples = [
+    "Who created Haystack AI?",
```
Comment (bilgeyucel) [inline integrations/apify.md:232]:
```suggestion
    "Who created Haystack?",
```

File: integrations/apify.md [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+# Components
+print("Initializing components...")
+document_store = InMemoryDocumentStore()
+
+docs_embedder = OpenAIDocumentEmbedder(api_key=Secret.from_token(openai_api_key))
+text_embedder = OpenAITextEmbedder(api_key=Secret.from_token(openai_api_key))
+retriever = InMemoryEmbeddingRetriever(document_store)
+generator = OpenAIGenerator(model="gpt-3.5-turbo")
+
+# Load documents from Apify
```
Comment (bilgeyucel) [inline integrations/apify.md:188]:
```suggestion
# Load documents from Apify
print("Crawling and indexing documents...")
print("You can visit https://console.apify.com/actors/runs to monitor the progress")
```

File: integrations/apify.md [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+# Now, connect the components to each other
+pipe.connect("embedder.embedding", "retriever.query_embedding")
+pipe.connect("retriever", "prompt_builder.documents")
+pipe.connect("prompt_builder", "llm")
+
+question = "What is haystack?"
+
+print("Running pipeline ... crawling will take some time ...")
+print("You can visit https://console.apify.com/actors/runs to monitor the progress")
+
```
Comment (bilgeyucel) [inline integrations/apify.md:224]: Crawling is not a part of this pipeline, it's already handled previously

```suggestion
print("Running pipeline ... ")
```

File: integrations/apify.md [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+  - name: apify
+    socials:
+      github: apify
+      twitter: apify
+      linkedin: apify
+pypi: https://pypi.org/project/apify-haystack
+repo: https://github.com/apify/apify-haystack
+type: Data Ingestion
+report_issue:  https://github.com/apify/apify-haystack/issues
+logo: /logos/apify.png
```
[the remark this answers is not in the document]
Comment (jirispilka) [reply integrations/apify.md:15]: > do you have logo alternatives? The logo should be visible on the dark background

Thanks for raising this. I'm checking with the team for a better logo.

File: integrations/apify.md [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+
+- Load a dataset from Apify and convert it to a Haystack Document
+- Extract data from Facebook/Instagram and save it in the InMemoryDocumentStore
+- Crawl websites, scrape text content, and store it in the InMemoryDocumentStore
+- Retrieval-Augmented Generation (RAG): Extracting text from a website & question answering
+
+The integration implements the following components:
+- `ApifyDatasetLoader`: Load a dataset created by an Apify Actor
+- `ApifyDatasetFromActorCall`: Call an Apify Actor, load the dataset, and convert it to Haystack Documents
+- `ApifyDatasetFromTaskCall`: Call an Apify task, load the dataset, and convert it to Haystack Documents
```
[the remark this answers is not in the document]
Comment (jirispilka) [reply integrations/apify.md:54]: Good point. I have an example for the `ApifyDatasetLoader` and `ApifyDatasetFromActorCall`

I added the following line

The integration implements the following components (you can find their usage in these [examples](https://github.com/apify/apify-haystack/tree/main/src/apify_haystack/examples)):

File: integrations/apify.md (hunk 7 above)
Comment (jirispilka) [reply integrations/apify.md:65]: Excellent! Thank you

**Review by bilgeyucel — APPROVED** [review]
Thank you @jirispilka. This PR is good to go once we update the logo

File: integrations/apify.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (jirispilka) [reply integrations/apify.md:15]: Replaced the logo with a transparent background.

**Pull request merged**